src/main/java/org/jpeek/Report.java

```diff
@@ -237,11 +215,26 @@ private XML xml() throws IOException {
                 String.format("XSL not found: %s", name)
             );
         }
-        return new XSLDocument(
-            new TextOf(res).asString(),
-            Sources.DUMMY,
-            this.params
-        ).transform(this.skeleton);
+        return new XMLDocument(
```

Comment:
Just got curious. Would it really be necessary to create a new XML wrapper, if the documentation to [`Xembler.applyQuietly(...)`](https://www.xembly.org/apidocs-0.22/org/xembly/Xembler.html#applyQuietly(org.w3c.dom.Node)) explicitly states that the returned `Node` is the same object as supplied? Which in turn implies that the whole document is mutable.

Reply:
Apparently mutating the `Node` returned by `node()` with `Xembler` doesn’t work that way, since `node()` clones whatever DOM nodes the `XMLDocument` impl returned by `XSLDocument.transform(...)` contains. Constructing a new `XMLDocument` seems to be the way.